### PR TITLE
DOC: Updated docstrings related to DateTimeIndex. GH22459

### PR DESCRIPTION
**Pull request by eldritchideen**
- [ ] closes #xxxx
- [x] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

Updated `pandas.date_range` and `pandas.bdate_range` docstrings to resolve errors raised from `scripts/validate_docstrings.py` from #22459. Also added an example to `pandas.bdate_range`.

### CODE REVIEW

File: pandas/core/indexes/datetimes.py [hunk 1]
```diff
@@ -1765,6 +1768,15 @@ def bdate_range(start=None, end=None, periods=None, freq='B', tz=None,
...
     <http://pandas.pydata.org/pandas-docs/stable/timeseries.html#offset-aliases>`__.
 
+    Examples
+    --------
+    Note how the two weekend days are skipped in the result.
+
+    >>> pd.bdate_range(start='1/1/2018', end='1/08/2018')
+    DatetimeIndex(['2018-01-01', '2018-01-02', '2018-01-03', '2018-01-04',
+               '2018-01-05', '2018-01-08'],
+              dtype='datetime64[ns]', freq='B')
```
Comment (gfyoung) [inline pandas/core/indexes/datetimes.py]: Hmmm...that output formatting of the lines is a little weird there.  Should they be more aligned?

File: pandas/core/indexes/datetimes.py [hunk 2]
```diff
@@ -1723,37 +1724,39 @@ def bdate_range(start=None, end=None, periods=None, freq='B', tz=None,
...
+        are passed.
 
         .. versionadded:: 0.21.0
 
     closed : string, default None
         Make the interval closed with respect to the given frequency to
-        the 'left', 'right', or both sides (None)
+        the 'left', 'right', or both sides (None).
+    **kwargs
+        For compatibility. Has no effect on the result.
```
Comment (gfyoung) [inline pandas/core/indexes/datetimes.py]: Not sure we need to document `**kwargs`?

cc @WillAyd @datapythonista

**Comment by jreback** [comment]
@datapythonista

**Review by datapythonista — APPROVED** [review]
lgtm

**Comment by gfyoung** [comment]
@eldritchideen : Can you rebase onto `master` ?  That will fix the Circle CI failures.

**Comment by datapythonista** [comment]
@eldritchideen Can you please merge master into your PR? `git fetch upstream && git merge upstream/master && git push`

This will rerun the CI checks, and when everything is green we can merge this.

**Comment by datapythonista** [comment]
Rebased, merging on green.

**Comment by jreback** [comment]
thanks @eldritchideen

**Pull request merged**